plugins/module_utils/prerequisites.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Jan Kodera (jkodera@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import sys
+import re
+import importlib
+import importlib.metadata
+import json
+
+# This file exists because Ansible currently cannot declare dependencies on Python modules
+# when imported. When imported, it fails with Ansible-compatible message if requirements
+# listed below are not installed
+# see https://github.com/ansible/ansible/issues/62733 for more info about lack of req support
+
+#############################
+
+# 'module_name, package_name, version_requirements' triplets
+DEPENDENCIES = [
+    ("fusion", "purefusion", ">=1.0.11,<2.0")
+]
+
+#############################
+
+# returns tuple (MAJOR, MINOR, PATCH)
+def _parse_version(val):
+    # regexes for this were really ugly
+    try:
```

Comment:
full regex:
```^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$```

Reply:
Would make sense, but I have two objections: 1) Python does not exactly guarantee the packages will use semantic versioning, so the current parsing routines are a bit lenient in what they accept, the regex is pretty strict, 2) lets face it, the regex is absolutely horrible and unmaintainable 😄 I've explicitly tried to keep of as many regexes as possible as I too wrote a (simpler) regex to parse it at first and it was completely unreadable, this is the same. It may be official, but I don't want to have to touch that

Reply:
```
major = int(parts[0])
        minor = int(parts[1])
        if len(parts) > 2:
            patch = re.match(r"^\d+", parts[2])
            patch = int(patch.group(0))
        else:
```

Reply:
In your code you also don't accepts versions like 1.0.11a1

Reply:
Yeah, that was intentional as usually anything with and beyond non-digit chars is something like alpha/beta/-prerelease etc. is just a subversion and there is no easy way to compare that. However I just wanted a working approach, it is definitely not the best that could be. For sure raise a suggestion if you have any ideas how to make it better

Reply:
Essentially I've just wanted and wrote something that
a) will parse most package version strings that we need to use
b) can compare the most important parts (major, minor and maybe to some degree patch) to check if we have all features we need